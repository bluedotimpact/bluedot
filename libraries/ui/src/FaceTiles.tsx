import React from 'react';

interface FaceTilesProps {
  faces: Array<{
    src: string;
    alt: string;
  }>;
  maxDisplay?: number;
  size?: number;
  className?: string;
}

export const FaceTiles: React.FC<FaceTilesProps> = ({
  faces,
  maxDisplay = 4,
  size = 32,
  className = '',
}) => {
  const displayFaces = faces.slice(0, maxDisplay);
  // TODO: Both of these should be fetched from the backend in the future
  const remainingCount = 4000;

  return (
    <div className={`face-tiles flex items-center ${className}`}>
      <div className="face-tiles__container flex -space-x-3">
        {displayFaces.map((face) => (
          <div
            key={face.src}
            className="face-tiles__avatar rounded-full border border-white overflow-hidden"
            style={{ width: size, height: size }}
          >
            <img
              src={face.src}
              alt={face.alt}
              className="face-tiles__avatar-img size-full object-cover"
            />
          </div>
        ))}
        {remainingCount > 0 && (
          <div
            className="face-tiles__counter flex items-center justify-center bg-bluedot-lighter text-bluedot-normal font-semibold text-sm border border-bluedot-normal w-[66px] h-8 rounded-2xl px-2 py-4"
          >
            +{remainingCount}
          </div>
        )}
      </div>
    </div>
  );
};

export default FaceTiles;
