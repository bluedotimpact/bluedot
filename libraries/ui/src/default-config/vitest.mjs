/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
