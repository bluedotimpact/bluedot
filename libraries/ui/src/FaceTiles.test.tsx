import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import { FaceTiles } from './FaceTiles';

describe('FaceTiles', () => {
  const mockFaces = [
    { src: '/avatar1.jpg', alt: 'User 1' },
    { src: '/avatar2.jpg', alt: 'User 2' },
    { src: '/avatar3.jpg', alt: 'User 3' },
    { src: '/avatar4.jpg', alt: 'User 4' },
    { src: '/avatar5.jpg', alt: 'User 5' },
  ];

  test('renders default as expected', () => {
    const { container } = render(<FaceTiles faces={mockFaces} />);
    expect(container).toMatchSnapshot();
  });

  test('renders with custom maxDisplay', () => {
    const { container } = render(<FaceTiles faces={mockFaces} maxDisplay={3} />);
    expect(container).toMatchSnapshot();
  });

  test('renders with custom size', () => {
    const { container } = render(<FaceTiles faces={mockFaces} size={48} />);
    expect(container).toMatchSnapshot();
  });

  test('renders with custom className', () => {
    const { container } = render(<FaceTiles faces={mockFaces} className="custom-class" />);
    expect(container).toMatchSnapshot();
  });

  test('renders empty state', () => {
    const { container } = render(<FaceTiles faces={[]} />);
    expect(container).toMatchSnapshot();
  });
});
