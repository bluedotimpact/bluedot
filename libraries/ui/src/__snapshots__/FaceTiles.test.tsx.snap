// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FaceTiles > renders default as expected 1`] = `
<div>
  <div
    class="face-tiles flex items-center "
  >
    <div
      class="face-tiles__container flex -space-x-3"
    >
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 1"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar1.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 2"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar2.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 3"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar3.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 4"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar4.jpg"
        />
      </div>
      <div
        class="face-tiles__counter flex items-center justify-center bg-bluedot-lighter text-bluedot-normal font-semibold text-sm border border-bluedot-normal w-[66px] h-8 rounded-2xl px-2 py-4"
      >
        +
        4071
      </div>
    </div>
  </div>
</div>
`;

exports[`FaceTiles > renders empty state 1`] = `
<div>
  <div
    class="face-tiles flex items-center "
  >
    <div
      class="face-tiles__container flex -space-x-3"
    >
      <div
        class="face-tiles__counter flex items-center justify-center bg-bluedot-lighter text-bluedot-normal font-semibold text-sm border border-bluedot-normal w-[66px] h-8 rounded-2xl px-2 py-4"
      >
        +
        4071
      </div>
    </div>
  </div>
</div>
`;

exports[`FaceTiles > renders with custom className 1`] = `
<div>
  <div
    class="face-tiles flex items-center custom-class"
  >
    <div
      class="face-tiles__container flex -space-x-3"
    >
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 1"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar1.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 2"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar2.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 3"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar3.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 4"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar4.jpg"
        />
      </div>
      <div
        class="face-tiles__counter flex items-center justify-center bg-bluedot-lighter text-bluedot-normal font-semibold text-sm border border-bluedot-normal w-[66px] h-8 rounded-2xl px-2 py-4"
      >
        +
        4071
      </div>
    </div>
  </div>
</div>
`;

exports[`FaceTiles > renders with custom maxDisplay 1`] = `
<div>
  <div
    class="face-tiles flex items-center "
  >
    <div
      class="face-tiles__container flex -space-x-3"
    >
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 1"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar1.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 2"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar2.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User 3"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar3.jpg"
        />
      </div>
      <div
        class="face-tiles__counter flex items-center justify-center bg-bluedot-lighter text-bluedot-normal font-semibold text-sm border border-bluedot-normal w-[66px] h-8 rounded-2xl px-2 py-4"
      >
        +
        4071
      </div>
    </div>
  </div>
</div>
`;

exports[`FaceTiles > renders with custom size 1`] = `
<div>
  <div
    class="face-tiles flex items-center "
  >
    <div
      class="face-tiles__container flex -space-x-3"
    >
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User 1"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar1.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User 2"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar2.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User 3"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar3.jpg"
        />
      </div>
      <div
        class="face-tiles__avatar rounded-full border border-white overflow-hidden"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User 4"
          class="face-tiles__avatar-img size-full object-cover"
          src="/avatar4.jpg"
        />
      </div>
      <div
        class="face-tiles__counter flex items-center justify-center bg-bluedot-lighter text-bluedot-normal font-semibold text-sm border border-bluedot-normal w-[66px] h-8 rounded-2xl px-2 py-4"
      >
        +
        4071
      </div>
    </div>
  </div>
</div>
`;
